lib/types/bcachefs.nix

```
@@ -0,0 +1,112 @@
+{
+  config,
+  device,
+  diskoLib,
+  lib,
+  options,
+  parent,
+  ...
+}:
+{
+  options = {
+    type = lib.mkOption {
+      type = lib.types.enum [ "bcachefs" ];
+      internal = true;
+      description = "Type";
+    };
+    device = lib.mkOption {
+      type = lib.types.str;
+      default = device;
+      description = "Device to use";
+    };
+    filesystem = lib.mkOption {
+      type = lib.types.str;
+      description = "Name of the bcachefs filesystem this partition belongs to";
+    };
+    # These are passed as arguments to the device corresponding to this one in the invocation of the `bcachefs format` command
+    # in the bcachefs_filesystem type defined in bcachefs_filesystem.nix used to format the bcachefs filesystem that this device is a part of.
+    extraFormatArgs = lib.mkOption {
+      type = lib.types.listOf lib.types.str;
+      default = [ ];
+      description = "Extra arguments passed to the bcachefs format command";
+    };
+    # This value is passed to the `--label` option for the device corresponding to this one in the invocation of the `bcachefs format` command
+    # in the bcachefs_filesystem type defined in bcachefs_filesystem.nix used to format the bcachefs filesystem that this device is a part of.
+    label = lib.mkOption {
+      type = lib.types.str;
+      default = "";
+      description = "Label to use for this device";
```

Let's not hide valuable information
```suggestion
      description = ''
        Label to use for this device.
        This value is passed to the `--label` option for the device corresponding to this one in the invocation of the `bcachefs format` command in the bcachefs_filesystem type. It is used to format the bcachefs filesystem that this device is a part of.
      '';
```